### PR TITLE
add_spawn command & more asgarnia ice dungeon spawns

### PR DESCRIPTION
**Pull request by KevinSenez**
## What has been done?
This adds the `add_spawn #id` command for admins+; Mostly useful in a dev  environement.
- Example: `add_spawn 50` would spawn the king black dragon the players location, add it as a set spawn in the proper region file corresponding to the players location.

Also adds more spawns to the Asgarnia Ice Dungeon:
- Pirates
- Hobgoblins

## Has your code been documented?
- Not everything has been, but can go back and add comments if necessary.

### CODE REVIEW

File: game/plugins/src/main/kotlin/gg/rsmod/plugins/content/cmd/addspawn.plugin.kts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        player.message(failMessage, type = ChatMessageType.CONSOLE)
+        e.printStackTrace()
+    }
+}
+
+fun addSpawn(player: Player, id: Int) {
+    val string = "spawn_npc(npc = Npcs.${getName(id)}, x = ${player.tile.x}, z = ${player.tile.z}, walkRadius = 5, direction = Direction.${player.faceDirection})"
+    try {
+        val file = File("./game/plugins/src/main/kotlin/gg/rsmod/plugins/content/areas/spawns/spawns_${player.tile.regionId}.plugin.kts")
+        if (!file.exists())
```
Comment (alycii) [inline game/plugins/src/main/kotlin/gg/rsmod/plugins/content/cmd/addspawn.plugin.kts:58]: Please use brackets here :)

File: game/plugins/src/main/kotlin/gg/rsmod/plugins/content/cmd/addspawn.plugin.kts [hunk 2]
```diff
@@ -39,14 +46,24 @@ fun tryWithUsage(player: Player, args: Array<String>, failMessage: String, tryUn
     }
 }
 
+/**
+ * Adds the spawn to the proper region file corresponding to the players location.
+ */
 fun addSpawn(player: Player, id: Int) {
-    val string = "spawn_npc(npc = Npcs.${getName(id)}, x = ${player.tile.x}, z = ${player.tile.z}, walkRadius = 5, direction = Direction.${player.faceDirection})"
+    if (getName(id) == null)
```
Comment (alycii) [inline game/plugins/src/main/kotlin/gg/rsmod/plugins/content/cmd/addspawn.plugin.kts:53]: Please use brackets here :)

File: game/plugins/src/main/kotlin/gg/rsmod/plugins/content/cmd/addspawn.plugin.kts [hunk 3]
```diff
@@ -39,14 +46,24 @@ fun tryWithUsage(player: Player, args: Array<String>, failMessage: String, tryUn
...
+    val string = "spawn_npc(npc = ${if (getName(id) == null) id else "Npcs.${getName(id)}"}, x = ${player.tile.x}, z = ${player.tile.z}, walkRadius = 5, direction = Direction.${player.faceDirection})"
     try {
         val file = File("./game/plugins/src/main/kotlin/gg/rsmod/plugins/content/areas/spawns/spawns_${player.tile.regionId}.plugin.kts")
         if (!file.exists())
             createSpawnFile(file)
         val writer = BufferedWriter(FileWriter(file, true))
-        writer.write("$string //Added by command by user: ${Misc.formatForDisplay(player.username)}")
+
+        val lines = Files.readAllLines(Paths.get(file.path))
+        if (lines.last().isNotEmpty() || lines.last().isNotBlank())
```
Comment (alycii) [inline game/plugins/src/main/kotlin/gg/rsmod/plugins/content/cmd/addspawn.plugin.kts:63]: Please use brackets here :)

(Okay, we should be never be doing one line `if() ->` without brackets, as that is the proper convention for the source code.)

File: game/plugins/src/main/kotlin/gg/rsmod/plugins/content/cmd/addspawn.plugin.kts [hunk 4]
```diff
@@ -39,14 +46,24 @@ fun tryWithUsage(player: Player, args: Array<String>, failMessage: String, tryUn
...
+            writer.newLine()
+
+        writer.write(string)
         writer.newLine()
         writer.close()
         player.message("Added Spawn for NPC=<col=FFFF00>$id</col> [Region=<col=FFFF00>${player.tile.regionId}</col> - Coords=<col=FFFF00>${player.tile.x}, ${player.tile.z}, ${player.tile.height}</col>]", type = ChatMessageType.CONSOLE)
```
Comment (alycii) [inline game/plugins/src/main/kotlin/gg/rsmod/plugins/content/cmd/addspawn.plugin.kts:69]: Might be better to print this as a command console message? Not required to pull, but better aesthetics

**Review by alycii — CHANGES_REQUESTED** [review]

**Comment by KevinSenez** [comment]
Resolved Issues 👌🏼

File: game/plugins/src/main/kotlin/gg/rsmod/plugins/content/cmd/addspawn.plugin.kts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        e.printStackTrace()
+    }
+}
+
+/**
+ * Adds the spawn to the proper region file corresponding to the players location.
+ */
+fun addSpawn(player: Player, id: Int) {
+    if (getName(id) == null) {
+        player.message("ERROR: val not found corresponding with given id; reverting to using supplied id instead.")
```
Comment (alycii) [inline game/plugins/src/main/kotlin/gg/rsmod/plugins/content/cmd/addspawn.plugin.kts:54]: This should be a console message :3

**Review by alycii — COMMENTED** [review]
chore: changed error message to console type

**Review by alycii — APPROVED** [review]
Thanks for resolving the issues :)

**Pull request merged**